config/Readme.md

```diff
@@ -0,0 +1,319 @@
+> Any validation and used default value is set on the called placed and is hard to detected by deploying and reuse on other places.
+
+## History and legacy tech stack
+
+### FeatherJS and Express
+
+Our legacy stack uses featherJS <https://docs.feathersjs.com/api/configuration.html>.
+It is embedded in the express application.
+
+Express and featherJS use the environment variable as the default behavior.
+NODE_ENV=production|test|default
+It is extended over featherJS and directly matched to
+./config/
+    default.json
+    test.json
+    prodcution.json (deprecated)
+    development.json (matching added by us)
+
+Default is (like it is expected) as default and is overwritten by added environment variables from test.json, or production.json.
+
+## Current and NestJS solutions
+
+### @hpi-schul-cloud/commons
+
+``` javascript
+    const { Configuration } = require('@hpi-schul-cloud/commons');
+
+    const url = Configuration.get('FILES_STORAGE__SERVICE_BASE_URL');
+```
+
+It is used for parsing any environment value that is added in ./config/default.schema.json.
+It is overridden with values in default.json.
+The default.json is also overridden by development.json (NODE_ENV==='default'), or test.json. (NODE_ENV==='test').
+
+For legacy featherJS stack, or legacy client it is the only solution that should be used.
+For newer nestjs stack we use it for parsing values in the right order, but map it to the nestjs based solution.
+> Look to the topic nestjs in this file for more information.
+
+The vue client we pass this values over an api endpoint.
+> Look to the topic "Passing configuration to vue client" in this file
+
+``` javascript
+    let configBefore;
+
+    before(() => {
+        configBefore = Configuration.toObject({ plainSecrets: true });
+
+        Configuration.set('ENVIRONMENT_NAME', 'fake.value');
+    });
+
+    after(async () => {
+        Configuration.reset(configBefore);
+    });
+
+```
+
+### ./config/global.js (deprecated)
+
+To collect and cleanup all existing process.env.XXX calls in code, it exists a step, where all environments variables are moved to global.js file.
+This should not be used anymore and cleaned up.
+
+Feel free to move variables to default.schema.json.
+
+### ./config/production.js (deprecated)
+
+This config values should not be used anymore.
+
+The default.schema.json and default.json represent the default values that should be set in all production systems.
+All other production values are added over autodeployment configurations.
+
+### ./config/default.schema.json
+
+We want to move any environment variable to this file for now.
+> Please add a discription and if possible default values to it.
+> Any default values that are set on this files should be for production systems. They can be overridden with autodeployment configurations.
+
+It make sense to cluster variables with same context. The situation can be different, but look like it is used atm.
```

Comment:
"The situation can be different, but look like it is used atm." Den Satz verstehe ich leider im Zusammenhang nicht so richtig.